express-delete/index.js

```diff
@@ -0,0 +1,46 @@
+const express = require('express');
+
+const grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+};
+
+const app = express();
+
+app.listen(3000, () => {
+  console.log('Listening on port 3000!');
+});
+
+app.get('/api/grades', (req, res) => {
+  const gradesArray = [];
+  for (let id in grades) {
+    gradesArray.push(grades[id]);
+  }
+  res.json(gradesArray);
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  for (let deleteId in grades) {
+    if (req.params.id === deleteId) {
+      delete grades[deleteId];
+      break;
+    }
+  }
```

Comment:
You can do this without a loop since you already have the property key in `req.params.id`.